src/loaders/stm_load.c

```diff
@@ -71,15 +95,17 @@ static int stm_test(HIO_HANDLE * f, char *t, const int start)
 	hio_seek(f, start + 20, SEEK_SET);
 	if (hio_read(buf, 1, 8, f) < 8)
 		return -1;
-	if (memcmp(buf, "!Scream!", 8) && memcmp(buf, "BMOD2STM", 8) && memcmp(buf, "WUZAMOD!", 8))
-		return -1;
```

Comment:
Ok, I see what Saga Musix meant by removing validation on the tracker name check—it's essentially this format's magic string. I also think removing this entirely is a bad idea. Aside from `PCSTV` and `SWavePro` and any case sensitivity issues is there anything else missing from this check? I think a list of correct strings here is a better idea than allowing anything. I'm not convinced just a check to make sure these are ASCII chars (like Saga Musix suggested) is a good way to handle this either, since, for example, Soundtracker 15-instrument .MODs will have ASCII chars here too.

Reply:
We don't need to bring the format check back as-is. With thorough enough header validation (see `ValidateHeader` in OpenMPT's Load_stm.cpp) you can be reasonably sure that something is really an STM file. There is more validation happening outside that function too in OpenMPT, e.g. if the order list contains any item above 63 that is not 99 or 255, the file is rejected as well.

Reply:
STM has no format magic string, just a field that contains some random text... or doesn't.

Reply:
While it may be technically possible to have completely random strings, only a handful of strings are found in the wild, that's what matters. It's not like most module players are unable to play a majority of STM modules because of this. While it may not be important for a single-purpose program like st2play to do thorough validation of this string, rejecting any non-ASCII string is a really valuable for multi-format libraries like libxmp that are also used in combination with other media playing libraries. But to move forward, I think we can merge this and have a follow-up PR to address these issues later.

Reply:
Yes, STM format has very few traits, so I agree with idea of additinal string validation. It will reduce false positives without hardcoding.